src/components/main/compose/messages/mod.rs

```diff
@@ -322,11 +322,23 @@ pub fn Messages(cx: Scope<Props>) -> Element {
 
     let rg = cx.props.messaging.clone();
     let senders: Vec<DID> = messages.iter().map(|msg| msg.sender()).collect();
+    let senders2 = senders.clone();
     // messages has already been reversed
     let idx_range = 0..messages.len();
     let next_sender = idx_range.clone().map(|idx| senders.get(idx + 1));
     let prev_sender = idx_range.map(|idx| if idx == 0 { None } else { senders.get(idx - 1) });
 
+    // get profile pictures for all senders in the conversation and cache them
+    let mut profile_pictures = HashMap::new();
+    for sender in senders2 {
```

Comment:
Instead of cloning to `senders2` you could do 

```suggestion
    for sender in senders.iter() {
```

and clone `sender` where needed (although from a glance `senders` isnt needed either since one can grab the recipients from `Conversation` so that way youre not iterating over every single message just for say 2 or 3 recipieints but that could probably be in this pr or in a different pr)

Reply:
Thanks, that's a much better way of doing it

Yeah the logic for senders could be handled a little better, I'll create a new issue for it